### PR TITLE
documentation: Preview Features

### PR DESCRIPTION
**Pull request by mcandeia**
# Description

Add preview features dev docs

## Issue reference

Please reference the issue this PR will close: #5162 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: docs/development/feature-flags.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Feature Flags
+
+Feature Toggles (a.k.a [feature flags](https://martinfowler.com/articles/feature-toggles.html)) are a powerful way to change runtime behavior without changing the code itself. We encourage the use of the feature flags especially for preview features - explicit opt-in, and this document is focused on showing best practices/examples, how to configure and how they can be enabled when appropriate (i.e running e2e tests).
+
+![image](https://martinfowler.com/articles/feature-toggles/chart-4.png)
```
Comment (ItalyPaleAle) [inline docs/development/feature-flags.md:5]: I'd remove this since it may be copyrighted.

Comment (mcandeia) [reply docs/development/feature-flags.md]: Thanks for catching this @ItalyPaleAle.

> you're the license god for now on 💟

File: docs/development/feature-flags.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Feature Flags
```
Comment (artursouza) [inline docs/development/feature-flags.md:1]: Please, use the term "Preview features". This doc seems to target contributors, please, see how you can reference our existing docs for preview features to users: https://docs.dapr.io/operations/support/support-preview-features/

**Review by artursouza — CHANGES_REQUESTED** [review]

File: docs/development/feature-flags.md (hunk 2 above)
Comment (mcandeia) [reply docs/development/feature-flags.md:1]: It is already referenced https://github.com/dapr/dapr/pull/5163/files#diff-367b06a8bbbee7752557e1f54fe4cd71c3c76b5ed2b32cdcbd9eb1dd879f4c14R94,

is that ok for you? Accepting suggestion

Comment (mcandeia) [reply docs/development/feature-flags.md]: see if it looks better now @artursouza

**Pull request merged**